src/useCapabilitiesSafe.ts

```diff
@@ -0,0 +1,27 @@
+import { useAccount } from 'wagmi';
+import { useCapabilities } from 'wagmi/experimental';
+import type { UseCapabilitiesSafeParams, WalletCapabilities } from './types';
+
+export function useCapabilitiesSafe({
+  chainId,
+}: UseCapabilitiesSafeParams): WalletCapabilities {
+  const { isConnected } = useAccount();
+
+  const { data: capabilities, error } = useCapabilities({
+    query: { enabled: isConnected },
+  });
+
+  if (error || !capabilities || !capabilities[chainId]) {
+    return {
+      hasPaymasterService: false,
+      hasAtomicBatch: false,
+      hasAuxiliaryFunds: false,
+    };
+  }
+
+  return {
+    hasPaymasterService: capabilities[chainId]?.paymasterService?.supported,
+    hasAtomicBatch: capabilities[chainId]?.atomicBatch?.supported,
+    hasAuxiliaryFunds: capabilities[chainId]?.auxiliaryFunds?.supported,
```

Comment:
if these can be undefined, should they be cast to a boolean?

Reply:
should be fine per [this test](https://github.com/coinbase/onchainkit/pull/1181/files#diff-c09e9ab67f227b7b90d1763987dfcaa7479e344e463d7030c4c4d178ee4703f8R73-R83)?

Reply:
that just tests if !capabilities though right?  is it possible for useCapabilities to return some of the capabilities, but not all?

Reply:
good catch :) fixed in the latest diff!